aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionAuthorizer.java

```diff
@@ -0,0 +1,219 @@
+                    .put("host", getHost())
+                    .put("Authorization", oidcProvider.getLatestAuthToken());
+        } catch (JSONException jsonException) {
+            throw new ApiException(
+                    "Error constructing the authorization json for Open ID Connect.",
+                    jsonException, AmplifyException.TODO_RECOVERY_SUGGESTION
+            );
+        }
+    }
+
+    private JSONObject forIam(
+            AWSCredentialsProvider credentialsProvider,
+            GraphQLRequest<?> request,
+            boolean connectionFlag
+    ) throws ApiException {
+        final URI apiUrl = getRequestEndpoint(connectionFlag);
+        final String apiRegion = apiUrl.getAuthority().split("\\.")[2];
+        final String requestContent = getRequestContent(request, connectionFlag);
+
+        // Construct a request to be signed
+        DefaultRequest<?> canonicalRequest = new DefaultRequest<>("appsync");
+        canonicalRequest.addHeader("accept", "application/json, text/javascript");
+        canonicalRequest.addHeader("content-encoding", "amz-1.0");
+        canonicalRequest.addHeader("content-type", "application/json; charset=UTF-8");
+        canonicalRequest.setHttpMethod(HttpMethodName.valueOf("POST"));
+        canonicalRequest.setEndpoint(apiUrl);
+        canonicalRequest.setContent(new ByteArrayInputStream(requestContent.getBytes()));
+
+        // Sign with AppSync's SigV4 signer that also considers connection resource path
+        new AppSyncV4Signer(apiRegion, connectionFlag).sign(
+                canonicalRequest,
+                credentialsProvider.getCredentials()
+        );
+
+        // Extract header from signed request and return
+        Map<String, String> signedHeaders = canonicalRequest.getHeaders();
+        JSONObject authorization = new JSONObject();
+        try {
+            for (Map.Entry<String, String> headerEntry : signedHeaders.entrySet()) {
+                authorization.put(headerEntry.getKey(), headerEntry.getValue());
+            }
+        } catch (JSONException jsonException) {
+            throw new ApiException(
+                    "Error constructing the authorization json for AWS IAM.",
+                    jsonException, AmplifyException.TODO_RECOVERY_SUGGESTION
+            );
+        }
+        return authorization;
+    }
+
+    private String getHost() {
+        return Uri.parse(configuration.getEndpoint()).getHost();
+    }
+
+    private URI getRequestEndpoint(boolean connectionFlag) throws ApiException {
+        try {
+            String baseUrl = configuration.getEndpoint();
+            String connectionUrl = connectionFlag ? baseUrl + "/connect" : baseUrl;
+            return new URI(connectionUrl);
+        } catch (URISyntaxException uriException) {
+            throw new ApiException(
+                    "Error constructing canonical URI for IAM request signature",
+                    uriException, AmplifyException.TODO_RECOVERY_SUGGESTION
```

Comment:
Use the `REPORT_TO_AWS`-or-whatever one that @drochetti added

Reply:
Didn't know we had that! I'll change it, and add appropriate recovery suggestion everywhere else.